### ISSUE TITLE
Does typescript and others need to be in dependencies?

### ISSUE THREAD

**Issue by razor-x** [saul/demofile#112]
I am deploying this on AWS lambda and hitting function size limits (200MB+ because of typescript).  Normally I would think typescript and `@types` would be devDependencies (for builds) not normal dependencies. Searching though the code I do not see these imported anywhere.

Can you do a quick review of these and move any that are not imported into `devDependencies`? That would help me a lot. Thanks.

```json
  "dependencies": {
    "@types/bytebuffer": "^5.0.37",
    "@types/lodash": "^4.14.116",
    "@types/node": "^10.11.3",
    "bit-buffer": "0.0.3",
    "bytebuffer": "^5.0.0",
    "lodash": "^4.17.11",
    "long": "^3.0.3",
    "protobufjs": "^6.8.8",
    "ts-assert-exists": "^1.0.4",
    "typescript": "^3.1.1"
  }

**Comment by saul** [comment]
Published 1.0.5 with this fix

**Comment by razor-x** [comment]
Thanks for getting in in so quickly :checkered_flag: 

Unfortunately it seems the dependency `ts-assert-exists` has the same issue and has included several devDependencies including typescript as dependencies. https://github.com/iamakulov/ts-assert-exists/blob/master/package.json#L7-L11 

```
    "dependencies": {
        "tslint": "^5.10.0",
        "tslint-config-prettier": "^1.13.0",
        "typescript": "^2.8.3"
},
```

I ignored this one at first since I saw it was imported and used. The end result is my `node_modules` folder still includes the (very-large) typescript package. 

I've confirmed removing this package fixes the size (with `npm install --production`)

```
before
❯ du -sch node_modules                                                                                             
53M     node_modules                                                                                               
53M     total                                                                                                      

after

❯ du -sch node_modules
11M     node_modules
11M     total
```

The entire source of that package is

```typescript
// Based upon https://github.com/Microsoft/TypeScript/issues/8655
function assertExists<Type>(
    value: Type | null | undefined,
    messageToThrow?: string,
): Type {
    if (value !== undefined && value !== null) {
        return value;
    } else {
        throw new Error(
            messageToThrow || 'assertExists: The passed value doesn’t exist',
        );
    }
}

export default assertExists;
```

So this means for 15 lines of code we incur a 50MB overhead :laughing: 

I suggest simply including this module directly. I will open a PR with this suggestion.

**Comment by saul** [comment]
Released v1.0.6
